src/cbor.c

```diff
@@ -45,49 +46,62 @@ cbor_token_t cbor_parseToken(const uint8_t* buf, size_t size)
 	if (val < 24) {
 		result.width = 0;
 		result.value = val;
-		return result;
-	}
+		// return result;
+	} else {
+		// shift buffer
+		// Holds minimum value for a given byte-width.
+		// Anything below this is not canonical CBOR as
+		// it could be represented by a shorter CBOR notation
+		uint64_t limit_min;
+		switch (val) {
+		case 24:
+			ENSURE_AVAILABLE_BYTES(1 + 1);
+			result.width = 1;
+			result.value = u1be_read(buf + 1);
+			limit_min = VALUE_W1_UPPER_THRESHOLD;
+			break;
+		case 25:
+			ENSURE_AVAILABLE_BYTES(1 + 2);
+			result.width = 2;
+			result.value = u2be_read(buf + 1);
+			limit_min = VALUE_W2_UPPER_THRESHOLD;
+			break;
+		case 26:
+			ENSURE_AVAILABLE_BYTES(1 + 4);
+			result.width = 4;
+			result.value = u4be_read(buf + 1);
+			limit_min = VALUE_W4_UPPER_THRESHOLD;
+			break;
+		case 27:
+			ENSURE_AVAILABLE_BYTES(1 + 8);
+			result.width = 8;
+			result.value = u8be_read(buf + 1);
+			limit_min = VALUE_W8_UPPER_THRESHOLD;
+			break;
+		default:
+			// Values above 27 are not valid in CBOR.
+			// Exception is indefinite length marker
+			// but this has been handled separately.
+			THROW(ERR_UNEXPECTED_TOKEN);
+		}
 
-	// shift buffer
-	// Holds minimum value for a given byte-width.
-	// Anything below this is not canonical CBOR as
-	// it could be represented by a shorter CBOR notation
-	uint64_t limit_min;
-	switch (val) {
-	case 24:
-		ENSURE_AVAILABLE_BYTES(1 + 1);
-		result.width = 1;
-		result.value = u1be_read(buf + 1);
-		limit_min = VALUE_MIN_W1;
-		break;
-	case 25:
-		ENSURE_AVAILABLE_BYTES(1 + 2);
-		result.width = 2;
-		result.value = u2be_read(buf + 1);
-		limit_min = VALUE_MIN_W2;
-		break;
-	case 26:
-		ENSURE_AVAILABLE_BYTES(1 + 4);
-		result.width = 4;
-		result.value = u4be_read(buf + 1);
-		limit_min = VALUE_MIN_W4;
-		break;
-	case 27:
-		ENSURE_AVAILABLE_BYTES(1 + 8);
-		result.width = 8;
-		result.value = u8be_read(buf + 1);
-		limit_min = VALUE_MIN_W8;
-		break;
-	default:
-		// Values above 27 are not valid in CBOR.
-		// Exception is indefinite length marker
-		// but this has been handled separately.
-		THROW(ERR_UNEXPECTED_TOKEN);
+		if (result.value < limit_min) {
+			// This isn't canonical CBOR
+			THROW(ERR_UNEXPECTED_TOKEN);
+		}
 	}
 
-	if (result.value < limit_min) {
-		// This isn't canonical CBOR
-		THROW(ERR_UNEXPECTED_TOKEN);
+	if (CBOR_TYPE_NEGATIVE == result.type) {
```

Comment:
I think we don't need to parse CBOR_TYPE_NEGATIVE. It is OK to have it here, but it should probably be just commented out (to avoid increasing app size) if we ever needed it in the future. (AFAIK we are only parsing CBOR for some legacy Byron stuff, like extracting protocol magic, or something like that.) But the size difference is perhaps negligible, so I guess I am OK with having it here (especially if we can't easily comment it out in a single place, I've not really checked).